djangocms_versioning/cms_config.py

```diff
@@ -8,57 +8,36 @@
     register_version_admin_for_models,
     replace_admin_for_models,
 )
-from .models import BaseVersion
+from .versionable import VersionableList
 
 
 class VersioningCMSExtension(CMSAppExtension):
 
     def __init__(self):
-        self.version_models = []
-        self.content_to_version_models = {}
+        self.versionables = VersionableList()
 
     def handle_versioning_models_setting(self, cms_config):
-        """Check the versioning_models setting has been correctly set
+        """Check the versioning setting has been correctly set
         and add the models to the masterlist if all is ok
         """
-        # First check that versioning_models is correctly defined
-        if not hasattr(cms_config, 'versioning_models'):
+        # First check that versioning is correctly defined
+        if not hasattr(cms_config, 'versioning'):
             raise ImproperlyConfigured(
-                "versioning_models must be defined in cms_config.py")
-        if not isinstance(cms_config.versioning_models, collections.abc.Iterable):
+                "versioning must be defined in cms_config.py")
+        if not isinstance(cms_config.versioning, collections.abc.Iterable):
             raise ImproperlyConfigured(
-                "versioning_models not defined as a list")
-        for model in cms_config.versioning_models:
-            try:
-                is_versioning_model = issubclass(model, BaseVersion)
-            except TypeError:
-                raise ImproperlyConfigured(
-                    "elements in versioning_models must be model classes")
-            if not is_versioning_model:
-                raise ImproperlyConfigured(
-                    "models in versioning_models must inherit from BaseVersion")
-        # If no exceptions raised, we can now add the versioning models
+                "versioning not defined as a list")
+        cms_config.versioning = VersionableList(cms_config.versioning)
```

Comment:
Is this really necessary? I don't think it's good to be changing the class another app has provided on the fly like that. And I have some (very vague) scenarios in which this could go wrong. Atm versioning is a dependency of moderation and they actually do check our settings. This is because they need to check the app is not just enabled for moderation but also for versioning and they need to make sure that each content model that is registered for moderation has also been registered for versioning.
This means that if you change stuff in this class on the fly like that odd things could potentially happen (or even be dependent on what the order of versioning and moderation is in INSTALLED_APPS). It also means this is a breaking change for moderation ;) Although I don't think there's any way around that, but please be aware that they will probably murder us soon for breaking their code all the time ;) It would be a good idea to include them more in changes like this that could break stuff for them.

Reply:
True, changed to not do that.